article.md

```diff
@@ -0,0 +1,309 @@
+
+By the end of this tutorial, you’ll have had a fully functional web application that you can use to scrape job postings from the LinkedIn website.
+
+
+
+![alt_text](images/image1.png "image_tooltip")
+
+
+Let’s get started
+
+
+## Prerequisites
+
+
+Let's start by creating a new Crawlee for Python project with this command:
+
+
+```
+pipx run crawlee create linkedin-scraper
+```
+
+
+Select `PlaywrightCrawler` in the terminal when Crawlee asks for it.
+
+After installation, Crawlee for Python will create a boilerplate code for you. Redirect into the project folder and run this command to install dependencies.
+
+
+```
+poetry install
+```
+
+
+
+## Building the LinkedIn Scraper using Crawlee for Python
+
+In this section, we are going to be building the scraper using the Crawlee Python package. To learn more about Crawlee, check out their [documentation](https://crawlee.dev/python/docs/quick-start).
+
+
+### Inspecting the LinkedIn job Search Page
+
+Open LinkedIn on your web browser and sign out from the website (if you already have an account logged in). You should see an interface like this.
+
+
+![alt_text](images/image2.png "image_tooltip")
+
+
+Navigate to the jobs section, search for a job and location of your choice; copy the Uniform Resource Locator (URL).
+
+
+
+![alt_text](images/image3.png "image_tooltip")
+
+
+You should have something like this:
+
+`https://www.linkedin.com/jobs/search?keywords=Backend%20Developer&location=Canada&geoId=101174742&trk=public_jobs_jobs-search-bar_search-submit&position=1&pageNum=0` 
+
+
+We're going to focus on the search parameters, which is the part that goes after '?'. The keyword and location parameters are the most important ones for us. 
+
+The job title the user supplies will serve as input to the keyword parameter while the location the user supplies will go into the location parameter. Lastly, the geoId parameter will be removed while we keep the other parameters constant.
+
+We are going to be making changes to our `main.py` file. Copy and paste the code below in your `main.py` file.
+
+
+```
+from crawlee.playwright_crawler import PlaywrightCrawler
+from .routes import router                                     
+import urllib.parse
+
+async def main(title: str, location: str, data_name: str) -> None:
+    base_url = "https://www.linkedin.com/jobs/search"
+
+    # URL encode the parameters
+    params = {
+        "keywords": title,
+        "location": location,
+        "trk": "public_jobs_jobs-search-bar_search-submit",
+        "position": "1",
+        "pageNum": "0"
+    }
+    encoded_params = urllib.parse.urlencode(params)
+    encoded_url = f"{base_url}?{encoded_params}"
+
+    # Initialize the crawler
+    crawler = PlaywrightCrawler(
+        request_handler=router,
+    )
```

Comment:
This is kinda confusing, because we haven't created or shown the router file yet, or explained what it is. The readers might think this is a runnable example and they'll be left with errors without any explanation.

Please start with some intermediary runnable state, (aka console.log in request handler), guide the user to making sure the crawler logs correctly, and then tell them that we will replace it with routing.

Reply:
During installation of the crawlee package, a boilerplate code has been created that contains the router.py file

We are simply editing those files.

By the time the user is done with the setup section, he will have the boiler plate code and he will see the routes.py file

Reply:
Ah yes, forgot we're starting from the template. 👍 

Still, would be nice to show the user what the correct behavior is, so that they feel comfortable they've set up everything correctly.

Usually I do something like:

> When you run the code, you will see the following output:

[screenshot of output from terminal]

Reply:
Okay, I've updated that as well.

Please check now